lib/kafka/cluster.rb

```diff
@@ -206,13 +207,14 @@ def fetch_cluster_info
 
           return cluster_info
         rescue Error => e
-          @logger.error "Failed to fetch metadata from #{node}: #{e}"
+          broker_error = "Failed to fetch metadata from #{node}: #{e}"
+          @logger.error broker_error
         ensure
           broker.disconnect unless broker.nil?
         end
       end
 
-      raise ConnectionError, "Could not connect to any of the seed brokers: #{@seed_brokers.join(', ')}"
+      raise ConnectionError, "Could not connect to any of the seed brokers: #{@seed_brokers.join(', ')}: #{broker_error}"
     end
```

Comment:
Would it make more sense to include _all_ the errors, e.g.

```ruby
errors = @seed_brokers.shuffle.map {|node|
  begin
    # ...
  rescue Error => e
    @logger.error "..."
    [node, e]
  else
    return cluster_info
  ensure
    # ...
  end
}

error_description = errors.map {|node, exception| "- #{node}: #{exception}" }.join("\n")

raise ConnectionError, "Could not connect to any of the seed brokers:\n#{error_description}"
```

Reply:
@dasch sorry I was out of the loop here a bit. Pushed now.